src/include/index/index_defs.h

```diff
@@ -0,0 +1,200 @@
+
+enum class IndexKind {
+  FlatL2,
+  IVFFlat,
+  FlatPQ,
+  IVFPQ,
+  Vamana,
+  VamanaPQ,
+  NNDescent,
+  Last
+};
+
+[[maybe_unused]] static std::vector<std::string> index_kind_strings{
+    "FlatL2",
+    "IVFFlat",
+    "FlatPQ",
+    "IVFPQ",
+    "Vamana",
+    "VamanaPQ",
+    "NNDescent",
+    "Last"};
+
+[[maybe_unused]] static inline auto str(IndexKind kind) {
+  return index_kind_strings[static_cast<int>(kind)];
+}
+
+/******************************************************************************
+ * Static info for arrays associated with an index group
+ ******************************************************************************/
+
+// @todo C++20 should have constexpr std::string
+[[maybe_unused]] static std::string current_storage_version{"0.3"};
+
+// @todo Use enum for key rather than string?
+using StorageFormat =
+    std::map<std::string, std::map<std::string, std::filesystem::path>>;
+[[maybe_unused]] static StorageFormat storage_formats = {
+    {"0.1",
+     {
+         {"centroids_array_name", "centroids.tdb"},
+         {"index_array_name", "index.tdb"},
+         {"ids_array_name", "ids.tdb"},
+         {"parts_array_name", "parts.tdb"},
+         {"input_vectors_array_name", "input_vectors"},
+         {"external_ids_array_name", "external_ids"},
+         {"partial_write_array_dir", "write_temp"},
+         {"default_attr_filters", ""},
+         {"updates_array_name", "updates"},
+         {"support_timetravel", "false"},
+     }},
+    {"0.2",
+     {
+         {"centroids_array_name", "partition_centroids"},
+         {"index_array_name", "partition_indexes"},
+         {"ids_array_name", "shuffled_vector_ids"},
+         {"parts_array_name", "shuffled_vectors"},
+         {"input_vectors_array_name", "input_vectors"},
+         {"external_ids_array_name", "external_ids"},
+         {"partial_write_array_dir", "temp_data"},
+         {"default_attr_filters", "zstd"},
+         {"updates_array_name", "updates"},
+         {"support_timetravel", "false"},
+     }},
+    {"0.3",
+     {
+         // From external/data/pytest-170/test_ivf_flat_ingestion_f320/array
+         // (generated by test/test_ingestion.py on 2023-12-01)
+         // Arrays comprising the index (note that temp_data has these as well)
+         {"centroids_array_name", "partition_centroids"},
+         {"index_array_name", "partition_indexes"},
+         {"ids_array_name", "shuffled_vector_ids"},
+         {"parts_array_name", "shuffled_vectors"},
+
+         // temp_data is a group, not an array
+         {"partial_write_array_name", "temp_data"},
+
+         // In storage_formats.py but not part of group?
+         {"input_vectors_array_name", "input_vectors"},
+         {"external_ids_array_name", "external_ids"},
+         {"updates_array_name", "updates"},
+
+         // Not arrays
+         // @todo Where are these stored?
+         {"default_attr_filters", "zstd"},
+         {"support_timetravel", "true"},
+     }},
+};
```

Comment:
(Unsure) Why does this one not have `{"partial_write_array_dir", "temp_data"}`? Will this cause a crash if we try to access it? In `apis/python/src/tiledb/vector_search/storage_formats.py`, the 0.3 version still has it.

Reply:
This is fine.  "partial_write_array_name" is just a key for "temp_data".  I just gave all the keys consistent names (x_y_array_name).